src/storage/copier/column_chunk.cpp

```diff
@@ -54,17 +54,21 @@ void ColumnChunk::append(common::ValueVector* vector, common::offset_t startPosI
 
 void ColumnChunk::append(
     ValueVector* vector, offset_t startPosInChunk, uint32_t numValuesToAppend) {
-    assert(vector->dataType.getLogicalTypeID() == LogicalTypeID::ARROW_COLUMN);
-    auto chunkedArray = ArrowColumnVector::getArrowColumn(vector).get();
-    for (auto array : chunkedArray->chunks()) {
-        auto numValuesInArrayToAppend =
-            std::min((uint64_t)array->length(), (uint64_t)numValuesToAppend);
-        if (numValuesInArrayToAppend <= 0) {
-            break;
+    if (vector->dataType.getLogicalTypeID() == LogicalTypeID::ARROW_COLUMN) {
```

Comment:
I know this is not your code but I'm very confused about this code. What I see is that this function (I mean `ColumnChunk::append(ValueVector* vector, offset_t startPosInChunk, uint32_t numValuesToAppend)`, calls itself recursively but does not do any actual copying. Instead for example the below or above `ColumnChunk::append` do some copying (below one uses memcpy somewhere and the above one calls `copyVectorToBuffer`).

Do you know what's going on or can you ask Guodong?